File: RepoTransBench/target_projects/Python/C++/rbarrois_xworkflows/public_tests/test_public_compat.cpp
#include <gtest/gtest.h>
#include "xworkflows/base.h"

TEST(PublicCompat, StringTypeIsStdString) {
    EXPECT_TRUE(typeid(xworkflows::base::Workflow::static_type_name()).hash_code() == typeid(std::string).hash_code());
}

// Base Workflow has states mapped/subscripted
TEST(PublicCompat, BaseWorkflowHasStates) {
    xworkflows::base::Workflow AltCompWorkflow{
        {{"alpha", "Alpha"}, {"beta", "Beta"}},
        {{"ab", {"alpha"}, "beta"}},
        "alpha"
    };
    auto& workflow = AltCompWorkflow;
    EXPECT_TRUE(workflow.states.find("alpha") != workflow.states.end());
    EXPECT_EQ(workflow.states.at("beta").title, "Beta");
}

File: RepoTransBench/target_projects/Python/Rust/rbarrois_xworkflows/public_tests/test_public_compat.rs
use rbarrois_xworkflows::base;

#[test]
fn test_public_string_type_is_str() {
    // Just a sanity check in Rust
    let s = std::any::type_name::<base::Workflow>();
    assert!(s.contains("Workflow"));
}

#[test]
fn test_public_base_workflow_has_states() {
    // Custom workflow
    struct AltCompWorkflow;
    impl AltCompWorkflow {
        fn states() -> Vec<(&'static str, &'static str)> {
            vec![("alpha", "Alpha"), ("beta", "Beta")]
        }
        fn transitions() -> Vec<(&'static str, &'static str, &'static str)> {
            vec![("ab", "alpha", "beta")]
        }
        fn initial_state() -> &'static str { "alpha" }
    }
    let workflow = AltCompWorkflow;
    let names: Vec<_> = workflow.states().iter().map(|x| x.0).collect();
    assert!(names.contains(&"alpha"));
    assert_eq!(workflow.states()[1].1, "Beta");
}